sklearn/linear_model/_base.py

```diff
@@ -518,7 +527,16 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if self.positive:
+            if y.ndim < 2:
+                self.coef_ = optimize.nnls(X, y)[0]
```

Comment:
`self._residues` is a private attribute, and I am not sure why we store it, but for consistency, I would store it in the single target case as well.